### PR TITLE
[FEATURE] Add support for iNavigateTo

### PR DESCRIPTION
**Pull request by gembeconsulting**
This PR provides support for iNavigateTo step, that sets a new hash to the app under test to trigger navigation. Developers can create test scenarios for navigation and routing of their UI5 app.

PR contains test scenario "iNavigateTo changes Hash" to verify step behavior.

### CODE REVIEW

File: src/steps/iNavigateTo.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * This step will set a new hash to the app.
+ * The app should react to the navigation by providing proper routing.
+ *
+ * ToDo: Add [with <JSON>] to synopsis to pass data as json
+ *    regex: "(\\s+with\\s+(.+?))?"
+ *    parsing: var oObject = JSON.parse(sObject);
+ */
+module.exports = {
+    docs: {
+        description: "Performs navigation using hash.",
```
Comment (darksmo) [inline src/steps/iNavigateTo.js:11]: - Do not include "." for consistency. 
- Also I would say "Navigates within the application using inner-app route". We may extends with Fiori Launchpad intent-based navigation later.

File: src/steps/iNavigateTo.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        synopsis: "I navigate to <hash>",
```
Comment (darksmo) [inline src/steps/iNavigateTo.js:12]: - Use 'I navigate to ROUTE'  (see synopsis in all other actions. Parameters are capitalized, with square brackets if they are optional)

File: src/steps/iNavigateTo.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        examples: [
+            "I navigate to /Customers/4711",
+            "I navigate to /Customers/4711?&tab=person"
```
Comment (darksmo) [inline src/steps/iNavigateTo.js:15]: - Consider adding more examples, especially with multiple parameters. You can test the examples without additional code via `npm test`, and see if they match the regex.
- In ...4711?&tab=person, was the "&" on purpose? I thought example should come without & although there are URLs with ?& around...
- Also perhaps we should add more example with multiple parameters

File: src/steps/iNavigateTo.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        ]
+    },
+    regexp: new RegExp([
+        "^",                          //start of string
+        "I\\snavigate\\sto\\s",       //enty point
+        "([=a-zA-Z0-9\/\?\&]+){1}",   //hash with parameters
```
Comment (darksmo) [inline src/steps/iNavigateTo.js:21]: Would use a more expressive regex here: `"(/|((/[a-zA-Z0-9_]+)+([?]([&a-zA-Z0-9_]+?=[^&?]*?)+)?))"` . You don't need to escape all characters when using a character class see https://www.regular-expressions.info/refcharclass.html .

File: src/steps/iNavigateTo.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        "$"                           //end of string
+    ].join("")),
+    icon: "locate-me",
+    action: function(sHash) {
+        var oWaitForOptions, that, oHasherChanger;
+        that = this;
+
+        oHasherChanger = sap.ui.test.Opa5.getHashChanger();
```
Comment (darksmo) [inline src/steps/iNavigateTo.js:29]: - Avoid global. Use this.Opa5.getHashChanger() instead

**Review by darksmo — CHANGES_REQUESTED** [review]
Well done. Let's fix some points and then merge this!

**Comment by darksmo** [comment]
Also let's not forget to sign CLA - otherwise we can't merge.

**Comment by SebastianWolf-SAP** [comment]
This repository is no longer maintained and will be archived. Therefore we will close this pull request. 

Thank you for your understanding and best regards, 

Sebastian (SAP Open Source Program Office)

**Pull request closed without merging**